src/shared/resourceloader.h

```diff
@@ -0,0 +1,38 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef RESOURCELOADER_H
+#define RESOURCELOADER_H
+
+#include <QList>
+#include <QObject>
+
+class QQmlAbstractUrlInterceptor;
+
+class ResourceLoader final : public QObject {
+  Q_OBJECT
+  Q_DISABLE_COPY_MOVE(ResourceLoader)
+
+ public:
+  static ResourceLoader* instance();
+
+  ~ResourceLoader();
+
+  QString loadFile(const QString& fileName);
+  QString loadDir(const QString& dirName);
+
+  void addUrlInterceptor(QQmlAbstractUrlInterceptor* interceptor);
+  void removeUrlInterceptor(QQmlAbstractUrlInterceptor* interceptor);
+
+ private:
+  ResourceLoader(QObject* parent);
+
+ signals:
+  void cacheFlushNeeded();
```

Comment:
This is a signal that needs to be implemented by users of the loaded resources IIUC. When is it necessary? Please add a docstring here with the explanation as well.

Reply:
This is called when an interceptor is added or removed. We need to inform the entire app about this operation.